### PR TITLE
Remove Note from plainify output

### PR DESCRIPTION
**Pull request by TristanCacqueray**
See: https://github.com/srid/emanote/pull/529

TODO(@srid)

- [x] Add tests (#1) in `master` branch
- [x] Add `plainify` tests in this PR 
- [ ] Refactor

### CODE REVIEW

File: src/Commonmark/Extensions/WikiLink.hs [hunk 1]
```diff
@@ -263,7 +263,15 @@ wikilinkSpec =
  TODO: extend on top of plainify from heist-extra
 -}
 plainify :: [B.Inline] -> Text
-plainify = W.query $ \case
+plainify = W.query plainify' . W.walk (mapMaybe removeInlineNotes)
```
Comment (srid) [inline src/Commonmark/Extensions/WikiLink.hs:266]: What's the difference compared to just doing it as part of `query`?

image

**Comment by srid** [comment]
I'll try to initialize some tests (#1) in the `master` branch. Then, we can add tests for this change before merging it.

File: src/Commonmark/Extensions/WikiLink.hs (hunk 1 above)
Comment (TristanCacqueray) [reply src/Commonmark/Extensions/WikiLink.hs:266]: If I understand correctly, `query` always traverses the Note childs, while `walk` can be used to remove the Note from the AST.

**Comment by srid** [comment]
I noticed that plainify'ication of regular links are broken, hence that test is disabled. Let's address this in different PR.

**Comment by srid** [comment]
May need follow-up fix for https://github.com/srid/emanote/pull/529#issuecomment-2018733384

File: src/Commonmark/Extensions/WikiLink.hs (hunk 1 above)
Comment (srid) [reply src/Commonmark/Extensions/WikiLink.hs:266]: [commenting several months later]

I completed overlooked the accidental double `W.query` here. But, no worries - fixed in https://github.com/srid/commonmark-wikilink/pull/5.

**Pull request merged**